### PR TITLE
test(anaconda-ai): add e2e CLI tests for anaconda ai server lifecycle and negative cases

### PR DESCRIPTION
**Pull request by RidaZubair**
## Ticket
[AIC-3046](https://anaconda.atlassian.net/browse/AIC-3046)

### Summary
Extends the Anaconda AI CLI e2e test suite to cover server lifecycle commands and error handling.

## New tests added:

- anaconda ai servers --json — verifies the servers list returns a valid JSON array
- anaconda ai launch <model>/<quant> --detach — verifies a model server starts with running status, OpenAI-compatible URL, and the expected model filename
- anaconda ai stop <model>.gguf — verifies the server stops and reports stopped + Success
- anaconda ai stop <model>.gguf --rm — verifies the server is deleted and reports deleted + Success
- anaconda ai launch abc (invalid format) — verifies non-zero exit code and ValueError with the correct error message
- anaconda ai launch nonexistent-model/q4_k_m (unknown model) — verifies non-zero exit code and ModelNotFound error

### Supporting changes:

- Added launchModelCmd, stopModelCmd, stopAndRemoveModelCmd, launchModelInvalidFormatCmd, launchModelNotFoundCmd to cliCommands.ts
- Added run/verify method pairs for all new commands to AnacondaAiCli in anaconda-ai-cmds.ts
- Added ServerApi type to model-api.ts
- Used test.step() in the launch/stop/delete test for clear failure attribution

[AIC-3046]: https://anaconda.atlassian.net/browse/AIC-3046?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tests/e2e/specs/cli/anaconda-ai.spec.ts [hunk 1]
```diff
@@ -37,4 +43,45 @@ test.describe('Anaconda AI CLI Commands @anaconda-ai', () => {
...
 
+  test('anaconda ai launch, stop model command', async ({ anacondaAiCli }) => {
+    await test.step('launch model server', async () => {
+      const launchResult = await anacondaAiCli.runLaunchModelCommand(
+        DOWNLOAD_TEST_MODEL_NAME,
+        DOWNLOAD_TEST_MODEL_QUANTIZATION,
+      );
+      anacondaAiCli.verifyLaunchModelCommand(launchResult, DOWNLOAD_TEST_MODEL_NAME, DOWNLOAD_TEST_MODEL_QUANTIZATION);
+    });
+
```
Comment (inhaz1) [inline tests/e2e/specs/cli/anaconda-ai.spec.ts:54]: Can we add one more validation step after the launch command to run anaconda ai servers --json and verify that the launched server appears in the list with status: "running"?

It would be good to also assert the response data, for example:

    "server_id": "bge-base-en-v1.5_q4_k_m.gguf",
    "model": "bge-base-en-v1.5_q4_k_m.gguf",
    "status": "running"

This would help confirm not just the launch output, but also that the server is actually listed in the running state.

Comment (RidaZubair) [reply tests/e2e/specs/cli/anaconda-ai.spec.ts:54]: Thank you! Added as a new step.

**Review by inhaz1 — APPROVED** [review]
LGTM
